### ISSUE TITLE
DAL rework: Cannot create a user via the admin

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#2700]
To replicate:

Create a user with:
* Username: test
* Password: test
* Email: vidange@vidange.com
* Registration window: 2017-10-20 - 2030-01-01
* Role: default
* Access duration: 12 hours

When you click create, it fails with "Error! Unexpected error. See server-side logs for details."

And if you check the logs you have:
```
Oct 20 12:33:06 localhost httpd_admin: httpd.admin(2595) ERROR: [mac:unknown] Trying to save a NULL value in a non nullable field person.pid (pf::dal::validate_field)
```

**Comment by extrafu** [comment]
Can you please retry with bob/bob?

**Comment by julsemaan** [comment]
Here is another warning in httpd.admin.error I forgot to add to the description
```
Oct 20 12:33:06 localhost httpd_admin_err: [Fri Oct 20 12:33:06 2017] -e: Use of uninitialized value $new_pid in lc at /usr/local/pf/lib/pf/person.pm line 250.
```

**Comment by jrouzierinverse** [comment]
Fixed by #2700

**Comment by julsemaan** [comment]
fix confirmed
